scitran/data/data.py

```diff
@@ -402,23 +401,12 @@ def parse(path, filetype=None, load_data=False, ignore_json=False, debug=False,
     timezone = None
     if not ignore_json:  # if ignore_json=False, read json
         log.debug('inspecting %s for json' % path)
-        with tarfile.open(path) as archive:
-            # TODO: explicit name check for Metadata.json or metadata.json
-            # if ti.name.lower is metadata.json
-            for ti in archive:
-                try:
-                    json_data = json.loads(archive.extractfile(ti).read(), object_hook=util.datetime_decoder)
-                except (TypeError, KeyError, AttributeError, ValueError):
-                    pass
-                else:
-                    log.debug('json found, %s' % ti.name)
-                    timezone = json_data.get('timezone')
-                    if not filetype:  # if filetype not already specified, get from json, or raise exception
-                        filetype = json_data.get('filetype')
-                        log.debug('filetype from json: %s' % filetype)
-                    break
-            else:
-                raise DataError('expected filetype to be indicated in json file')
+        json_data = json.loads(zipfile.ZipFile(path).comment)
```

Comment:
Let's add a try/except here to not crash when the comment does not contain JSON.